From 56df0d3fb0259ec28380a9b98e6c9955ec8070c0 Mon Sep 17 00:00:00 2001
From: Tim Hartley <timothy.d.hartley@intel.com>
Date: Thu, 5 Sep 2013 09:57:09 -0700
Subject: Klocwork fix in PCG GL

BZ: 136398

A minor fix to dvmCompilerPcgGetSymbolAddress to fix a dereference
that may be null.

Category: device-enablement
Domain: AOSP-Dalvik-Compiler-CG
Origin: internal
Upstream-Candidate: no, needs rework

Change-Id: Ie2d0b0a045e9a51823a99319a76c7b3e0533a03b
Orig-MCG-Change-Id: Ib1817f4d74377727563020a2b0b297355fdffc4a
Signed-off-by: Tim Hartley <timothy.d.hartley@intel.com>
Reviewed-on: http://android.intel.com:8080/131450
Reviewed-by: Katkov, Serguei I <serguei.i.katkov@intel.com>
Reviewed-by: Kreitzer, David L <david.l.kreitzer@intel.com>
Reviewed-by: Lupusoru, Razvan A <razvan.a.lupusoru@intel.com>
Reviewed-by: Beyler, Jean Christophe <jean.christophe.beyler@intel.com>
Reviewed-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: lab_aqa <lab_aqa@intel.com>
Reviewed-by: Popov, Ivan G <ivan.g.popov@intel.com>
Tested-by: Chen, Dong-Yuan <dong-yuan.chen@intel.com>
Reviewed-by: Banerji, Udayan <udayan.banerji@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
Signed-off-by: Serguei Katkov <serguei.i.katkov@intel.com>
---
 vm/compiler/codegen/x86/pcg/Labels.cpp |    5 +++++
 1 files changed, 5 insertions(+), 0 deletions(-)

diff --git a/vm/compiler/codegen/x86/pcg/Labels.cpp b/vm/compiler/codegen/x86/pcg/Labels.cpp
index a05c6f7..e522a20 100644
--- a/vm/compiler/codegen/x86/pcg/Labels.cpp
+++ b/vm/compiler/codegen/x86/pcg/Labels.cpp
@@ -150,6 +150,11 @@ void* dvmCompilerPcgGetSymbolAddress (CGSymbol cgSymbol)
     //Paranoid
     assert (clientSymbol != 0);
 
+    if (clientSymbol == 0)
+    {
+        return (void*)0;
+    }
+
     //Return the address
     return clientSymbol->address;
 }
-- 
1.7.4.1

